Jenkinsfile.k8s

```diff
@@ -89,41 +96,71 @@ pipeline {
             ninja -j $NP
           '''
         }
-      } 
+      }
     }
 
-    stage('Prebuild tests') {
+    stage('Run tests') {
       options {timeout(time: 90, unit: 'MINUTES')}
       steps {
         container('dealii'){
           sh '''
+            # This export avoids a warning about
+            # a discovered, but unconnected infiniband network.
             export OMPI_MCA_btl=self,tcp
+
             cd /home/dealii/build-gcc-fast/tests
-            echo "Prebuilding tests..."
-	    ninja -k 0 tests || true
-	    '''
-	}
-      }
-    }
 
-    stage('Run tests') {
-      options {timeout(time: 90, unit: 'MINUTES')}
-      steps {
-        container('dealii'){
+            # Let ninja prebuild the test libraries and run
+            # the tests to create the output files in parallel. We
+            # want this to always succeed, because it does not generate
+            # useful output (we do this further down using 'ctest', however
+            # ctest can not run ninja in parallel, so this is the
+            # most efficient way to build the tests).
+            ninja -k 0 tests || true
```

Comment:
We used to have this in a separate step. This has the advantage that the output won't show up as only the last failed step is typically shown. How does that work now?

Reply:
It's part of the "Run tests" step, but since it can't fail, the ctest sub-step will be displayed (unfurled?) in the Jenkins BlueOcean interface. (If it failed.)

Reply:
It's part of the "Run tests" step, but since it can't fail, the ctest sub-step will be displayed (unfurled?) in the Jenkins BlueOcean interface. (If it failed.)

Reply:
I see now. I don't think I had realized quite what `ninja tests` does exactly.

Reply:
@tjhei: You can see the current output here: https://singleb.cse.ucdavis.edu/blue/organizations/jenkins/aspect/detail/PR-2604/41/pipeline
The prebuild output does only show up if you click on the first line of the tests stage. But the best place to see the test results is now here:
https://singleb.cse.ucdavis.edu/blue/organizations/jenkins/aspect/detail/PR-2604/41/tests
It will show all failed tests at the top, including their output, which is even better than before (where we had to search through the ctest output of all tests for the output of the failed tests).